src/main/scala/interrupts/Crossing.scala

```diff
@@ -56,11 +67,51 @@ class IntSyncCrossingSink(sync: Int = 3)(implicit p: Parameters) extends LazyMod
   }
 }
 
-object IntSyncCrossingSink
+object IntSyncAsyncCrossingSink
 {
   def apply(sync: Int = 3)(implicit p: Parameters) =
   {
-    val intsink = LazyModule(new IntSyncCrossingSink(sync))
+    val intsink = LazyModule(new IntSyncAsyncCrossingSink(sync))
+    intsink.node
+  }
+}
+
+class IntSyncSyncCrossingSink()(implicit p: Parameters) extends LazyModule
+{
+  val node = IntSyncSinkNode(0)
+
+  lazy val module = new LazyModuleImp(this) {
+    (node.in zip node.out) foreach { case ((in, edgeIn), (out, edgeOut)) =>
+      out := in.sync
+    }
+  }
+}
+
+object IntSyncSyncCrossingSink
+{
+  def apply()(implicit p: Parameters) =
+  {
+    val intsink = LazyModule(new IntSyncSyncCrossingSink())
+    intsink.node
+  }
+}
+
+class IntSyncRationalCrossingSink()(implicit p: Parameters) extends LazyModule
+{
+  val node = IntSyncSinkNode(1)
+
+  lazy val module = new LazyModuleImp(this) {
+    (node.in zip node.out) foreach { case ((in, edgeIn), (out, edgeOut)) =>
+      out := RegNext(in.sync)
```

Comment:
This never needs to be more than 1 register stage deep? I would've expected this to be implemented similarly to the above `class IntSyncAsyncCrossingSink(sync: Int = 3)` with configurable depth